File: archive-admin/src/types/tenant.ts
export interface Tenant {
    id: string
    uuid: string
    name: string
    slug: string
    domain?: string
    logo?: string
    status: 'active' | 'inactive' | 'suspended'
    plan: 'free' | 'starter' | 'professional' | 'enterprise'
    created_at: string
    updated_at: string
    settings?: TenantSettings
    stats?: TenantStats
}

export interface TenantSettings {
    theme?: {
        primary_color?: string
        logo_url?: string
        favicon_url?: string
    }
    features?: {
        comments_enabled?: boolean
        support_enabled?: boolean
        analytics_enabled?: boolean
    }
    limits?: {
        max_posts?: number
        max_users?: number
        max_storage_mb?: number
    }
}

export interface TenantStats {
    total_posts: number
    total_users: number
    total_categories: number
    storage_used_mb: number
}

export interface CreateTenantRequest {
    name: string
    slug: string
    domain?: string
    plan?: string
}

export interface UpdateTenantRequest {
    name?: string
    slug?: string
    domain?: string
    status?: string
    settings?: TenantSettings
}

File: cortex/tenant/tenant.go
package tenant

import (
	"time"

	"github.com/google/uuid"
)

const (
	StatusActive    = "active"
	StatusInactive  = "inactive"
	StatusSuspended = "suspended"
)

const (
	PlanFree         = "free"
	PlanStarter      = "starter"
	PlanProfessional = "professional"
	PlanEnterprise   = "enterprise"
)

type Tenant struct {
	ID        int                    `json:"id" db:"id"`
	UUID      uuid.UUID              `json:"uuid" db:"uuid"`
	Name      string                 `json:"name" db:"name"`
	Slug      string                 `json:"slug" db:"slug"`
	Domain    *string                `json:"domain,omitempty" db:"domain"`
	Status    string                 `json:"status" db:"status"`
	Plan      string                 `json:"plan" db:"plan"`
	Settings  map[string]interface{} `json:"settings,omitempty" db:"settings"`
	Meta      map[string]interface{} `json:"meta,omitempty" db:"meta"`
	CreatedAt time.Time              `json:"created_at" db:"created_at"`
	UpdatedAt time.Time              `json:"updated_at" db:"updated_at"`
	Stats     *TenantStats           `json:"stats,omitempty"`
}

type TenantStats struct {
	TotalPosts      int `json:"total_posts"`
	TotalUsers      int `json:"total_users"`
	TotalCategories int `json:"total_categories"`
	StorageUsedMB   int `json:"storage_used_mb"`
}
